src/common/dateInput/DateInputLogic.tsx

```diff
@@ -0,0 +1,240 @@
+  monthInputName: string;
+  yearInputName: string;
+  dateComponent: React.FC<InputComponentProps>;
+  monthComponent: React.FC<InputComponentProps>;
+  yearComponent: React.FC<InputComponentProps>;
+  dateInputLabel: string;
+  monthInputLabel: string;
+  yearInputLabel: string;
+}
+
+function getDateComponents(date: Date | null): DateObject {
+  if (!date) {
+    return {};
+  }
+
+  const monthDate = date.getDate();
+  const month = date.getMonth();
+  const year = date.getFullYear();
+
+  return {
+    date: monthDate,
+    month,
+    year,
+  };
+}
+
+function makeDate(date: DateObject): Date | null {
+  if (date.year && date.month && date.date) {
+    return new Date(date.year, date.month, date.date, 0, 0, 0, 0);
+  }
+
+  return null;
+}
+
+function getCharacterCount(number: number): number {
+  return number.toString().length;
+}
+
+function sanitizeInput(content: unknown) {
+  return String(content)
+    .split('')
+    .filter(character => DATE_INPUT_ALLOWED_CHARACTERS.includes(character))
+    .join('');
+}
+
+function cutToLength(string: string, length: number) {
+  if (string.length <= length) {
+    return string;
+  }
+
+  return string.slice(0, length);
+}
+
+function parseNumberFromValue(value: string, maxCharacterCount = 2): number {
+  // Removes all other characters than numbers from the result. This is
+  // a last fallback which makes negative numbers positive and removes
+  // possible characters from the output. The behavior of
+  // input[type="number"] is not consistent between platforms.
+  const sanitized = sanitizeInput(value);
+  const maxSized = cutToLength(sanitized, maxCharacterCount);
+
+  return Number(maxSized);
+}
+
+function DateInputLogic({
+  onChange,
+  value,
+  isInvalid,
+  wrapper,
+  dateInputId,
+  monthInputId,
+  yearInputId,
+  dateInputName,
+  monthInputName,
+  yearInputName,
+  dateComponent,
+  monthComponent,
+  yearComponent,
+  dateInputLabel,
+  monthInputLabel,
+  yearInputLabel,
+}: Props) {
+  const currentDate = getDateComponents(value);
+  const [cachedDate, setCachedDate] = React.useState<DateObject>(currentDate);
+  const monthInputRef = React.useRef(null);
+  const yearInputRef = React.useRef(null);
+
+  const updateDate = (
+    date: number | null,
+    month: number | null,
+    year: number | null
+  ) => {
+    setCachedDate(previousDate => {
+      const nextCachedDate = {
+        date: defaultTo(date, previousDate.date),
+        month: defaultTo(month, previousDate.month),
+        year: defaultTo(year, previousDate.year),
+      };
+      const nextDate = makeDate(nextCachedDate);
+
+      if (nextDate) {
+        onChange(nextDate);
+      }
+
+      return nextCachedDate;
+    });
+  };
+
+  const focusWhen = (
+    condition: boolean,
+    element: React.RefObject<HTMLInputElement>
+  ) => {
+    if (condition && element.current) {
+      element.current.focus();
+    }
+  };
+
+  const handleDateChange = (event: React.FormEvent<HTMLInputElement>) => {
```

Comment:
Is this handling the whole date like 2020-12-25 OR the day of month? Maybe you can rename to handleDayChange in that case.

Reply:
With JS Dates, date = day of month and day = day of week. I don't know how dogmatically this should be followed and where is the correct point to switch naming. My rationale here was that this input control's the date value of a JS Date, so we want to reflect that in the naming as well.

https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Date/getDate
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Date/getDay

Reply:
Oh, thank you for this input, I should have known this!